roles/infrastructure/tasks/database.yml

```diff
@@ -0,0 +1,99 @@
+    resource_group: "{{ resource_group }}"
+    resource_name: "aap-infrastructure-{{ deployment_id }}-db"
+    resource_type: 'flexibleServers'
+    api_version: '2022-12-01'
+    body: "{{ lookup('template', 'database_config.j2') }}"
+  tags:
+    - database
+
+- name: Wait for PostgreSQL Flexible Server to finish provisioning
+  azure.azcollection.azure_rm_resource_info:
+    provider: 'DBforPostgreSQL'
+    resource_group: "{{ resource_group }}"
+    resource_name: "aap-infrastructure-{{ deployment_id }}-db"
+    resource_type: 'flexibleServers'
+    api_version: '2022-12-01'
+  register: result
+  until: result.response[0].properties is defined and result.response[0].properties.state == 'Ready'
+  retries:
+    20
+  delay:
+    60
+  tags:
+    - database
+
+- name: Create database for Controller
+  azure.azcollection.azure_rm_resource:
+    idempotency: true
+    resource_name: "aap-infrastructure-{{ deployment_id }}-db"
+    resource_group: "{{ resource_group }}"
+    provider: 'DBforPostgreSQL'
+    resource_type: "flexibleServers"
+    subresource:
+      - name: "awx"
+        type: "databases"
+    body: {"properties": {"charset": "utf8", "collation": "en_US.utf8"}}
+  tags:
+    - database
+
+- name: Create database for Hub
+  azure.azcollection.azure_rm_resource:
+    idempotency: true
+    resource_name: "aap-infrastructure-{{ deployment_id }}-db"
+    resource_group: "{{ resource_group }}"
+    provider: 'DBforPostgreSQL'
+    resource_type: "flexibleServers"
+    subresource:
+      - name: "hub"
+        type: "databases"
+    body: {"properties": {"charset": "utf8", "collation": "en_US.utf8"}}
+  tags: hub
+  when: "'hub' in ansible_run_tags"
+
+- name: Create database for EDA
+  azure.azcollection.azure_rm_resource:
+    idempotency: true
+    resource_name: "aap-infrastructure-{{ deployment_id }}-db"
+    resource_group: "{{ resource_group }}"
+    provider: 'DBforPostgreSQL'
+    resource_type: "flexibleServers"
+    subresource:
+      - name: "eda"
+        type: "databases"
+    body: {"properties": {"charset": "utf8", "collation": "en_US.utf8"}}
+  tags: eda
+  when: "'eda' in ansible_run_tags"
+
+- name: Get list of enabled extensions
+  azure.azcollection.azure_rm_resource_info:
+    provider: 'DBforPostgreSQL'
+    resource_group: "{{ resource_group }}"
+    resource_name: "aap-infrastructure-{{ deployment_id }}-db/configurations/azure.extensions"
+    resource_type: 'flexibleServers'
+    api_version: '2022-12-01'
+  register: postgresql_info
+  tags:
+    - database
+
+- name: Enable extensions
+  azure.azcollection.azure_rm_resource:
+    idempotency: true
+    provider: 'DBforPostgreSQL'
+    resource_group: "{{ resource_group }}"
+    resource_name: "aap-infrastructure-{{ deployment_id }}-db/configurations/azure.extensions"
+    resource_type: 'flexibleServers'
+    body: "{{ postgresql_info.response[0] | combine({'properties': {'value': (extensions | default('') | join(',')), 'source': 'user-override'}}) }}"
```

Comment:
Pretty sure we need `hstore` here like we have on the managed side.
https://github.com/ansible/aap-azurerm/blob/main/modules/postgreSQL/flexibleServers.bicep#L44

Reply:
```suggestion
    body: "{{ postgresql_info.response[0] |
      combine({'properties': {'value': (infrastructure_database_extensions | default('') | join(',')), 'source': 'user-override'}}) }}"
```